### ISSUE TITLE
Enforce GCC version

### ISSUE THREAD

**Issue by oleg-alexandrov** [CGAL/cgal#4409]
## Issue Details

CGAL 5 fails to compile with GCC 5 giving a lot of errors. Since in CHANGES.md it is written that only  Gnu g++ 6.3 is supported, the CMakeLists.txt must check for this and error out if the correct version is not found.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 
* Compiler: gcc 5
* Release or debug mode: Release
* Specific flags used (if any):
* CGAL version: 5
* Boost version: 
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by lrineau** [comment]
Even without being officially supported, g++ version 5.0 and later should be fine with CGAL-5.0, and that is why CMake allows it. What sort of errors do you have?

**Comment by lrineau** [comment]
If you choose a gcc compiler before 5.0, CMake will give an error like this one:
```
CMake Error in AABB_tree/examples/AABB_tree/CMakeLists.txt:
  The compiler feature "cxx_decltype_auto" is not known to CXX compiler

  "GNU"

  version 4.8.5.
```

**Comment by oleg-alexandrov** [comment]
Thank you for your very prompt response.

I am getting the error: 

CGAL/include/CGAL/Lazy.h:831:17: error: 'remove_cv_t' is not a member of 'std'
   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
                 ^

I had to specify in the cmake file:

set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

for it to work. Then I am getting the link error:

(.text+0x8ac): undefined reference to `__gmpn_copyi'

This is with:

gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 

In short, the documentation could be more clear about what compiler to use and how to invoke it. 

Thanks again. 

I also see that this was addressed at: https://github.com/CGAL/cgal/issues/3249

In short, CGAL is not header-only as the documentation states, as some dependencies must still be linked.

**Comment by lrineau** [comment]
My guess, @oleg-alexandrov, is that you are not using CGAL CMake scripts to compile your program. Because C++14 is automatically implied as soon as you link the CMake target `CGAL::CGAL`. As for `__gmpn_copyi`, it comes from `libgmp.so`, that is also automatically linked with all CMake targets that are linked with `CGAL::CGAL`. If you have your own build system (not using CMake), then make sure the Ubuntu package `libmpfr-dev` is installed, and add `-lgmp -lmpfr` to the linker flags.

**Comment by oleg-alexandrov** [comment]
Thank you. It is good to know GCC 5 can still be used with CGAL even if not officially supported.
